homeassistant/components/homematicip_cloud/services.py

```diff
@@ -324,6 +344,18 @@ async def _async_reset_energy_counter(hass: HomeAssistant, service: ServiceCall)
                     await device.reset_energy_counter()
 
 
+async def _async_set_home_cooling_mode(hass: HomeAssistant, service: ServiceCall):
+    """Service to set the cooling mode."""
+    cooling = service.data[ATTR_COOLING]
+
+    if hapid := service.data.get(ATTR_ACCESSPOINT_ID):
+        if home := _get_home(hass, hapid):
```

Comment:
Should we raise `ServiceValidationError` here if it can't find the "home"?
Right now if an incorrect id is provided nothing happens. With `ServiceValidationError` we could give a good response to the user.

Reply:
That should probably happen in the `_get_home` function then? Currently only a log entry is added there.

>     _LOGGER.info("No matching access point found for access point id %s", hapid)

I could change that, but that would mean I would change the existing behaviour for the rest of the services. Or do you want me add that specifically for this new service?

Reply:
Should I just replace that line with 
```
raise ServiceValidationError(
    translation_domain=DOMAIN,
    translation_key="accesspoint_not_found",
    translation_placeholders={"id": hapid},
)
```
And add the exception to the strings. Or do you want me to create a new PR for this (as it changes the rest of the services also)

Reply:
Yes, I think it makes sense to change the logging to using `ServiceValidationError`. Then we will also fix this for all action calls at the same time.
Don't forget to update the test.

Reply:
I added the exception + test